### PR TITLE
RHCLOUD-6055 - always update stale_timestamp from next reporter

### PR DESCRIPTION
**Pull request by tahmidefaz**
Jira: https://projects.engineering.redhat.com/browse/RHCLOUD-6055

### CODE REVIEW

**Comment by tahmidefaz** [comment]
I think I got everything. Lmk if I need to change/clarify things.

**Comment by jharting** [comment]
Let's make sure platform docs are updated before we release this

File: test_unit.py [hunk 1]
```diff
@@ -1425,30 +1425,7 @@ def test_updated_with_same_reporter(self):
...
-        host = self._make_host(stale_timestamp=old_stale_timestamp, reporter=reporter)
-
-        new_stale_timestamp = datetime.now(timezone.utc) + timedelta(days=1)
-        host._update_stale_timestamp(new_stale_timestamp, reporter)
-
-        self.assertEqual(old_stale_timestamp, host.stale_timestamp)
-        self.assertEqual(reporter, host.reporter)
-
-    def test_not_updated_with_different_reporter(self):
+    def test_always_updated_with_different_reporter(self):
```
Comment (Glutexo) [inline test_unit.py:1428]: We don’t need two separate tests when the condition is now gone. We can keep all the combinations just as we did in the API test for regressions, or just check once with any values.

```python
from itertools import product

    def test_always_updated(self):
        old_stale_timestamp = datetime.now(timezone.utc) + timedelta(days=2)
        old_reporter = "old reporter"
        stale_timestamps = (old_stale_timestamp - timedelta(days=1), old_stale_timestamp - timedelta(days=2))
        reporters = (old_reporter, "new reporter")
        for new_stale_timestamp, new_reporter in product(stale_timestamps, reporters):
            with self.subTest(stale_timestamps=new_stale_timestamp, reporter=new_reporter):
                host = self._make_host(stale_timestamp=old_stale_timestamp, reporter=old_reporter)

                new_stale_timestamp = datetime.now(timezone.utc) + timedelta(days=2)
                host._update_stale_timestamp(new_stale_timestamp, new_reporter)

                self.assertEqual(new_stale_timestamp, host.stale_timestamp)
                self.assertEqual(new_reporter, host.reporter)
```

```
    def test_updated_when_set(self):
        host = self._make_host(stale_timestamp=datetime.now(timezone.utc), reporter="old reporter")

        new_stale_timestamp = datetime.now(timezone.utc)
        new_reporter = "new reporter"
        host._update_stale_timestamp(new_stale_timestamp, new_reporter)

        self.assertEqual(new_stale_timestamp, host.stale_timestamp)
        self.assertEqual(new_reporter, host.reporter)
```

I like the exhaustive version better. More tests is usually better than fewer tests. It also serves as a documentation of the now removed logic.

Comment (tahmidefaz) [reply test_unit.py]: @Glutexo I choose the first example

**Comment by tahmidefaz** [comment]
This is ready for a final review/merge

**Pull request merged**